libraries/chain/database.cpp

```diff
@@ -246,22 +246,46 @@ bool database::is_known_transaction( const transaction_id_type& id )const
    return trx_idx.find( id ) != trx_idx.end();
 } FC_CAPTURE_AND_RETHROW() }
 
-block_id_type database::get_block_id_for_num( uint32_t block_num )const
+block_id_type database::find_block_id_for_num( uint32_t block_num )const
 {
    try
    {
+      if( block_num == 0 )
+         return block_id_type();
+
+      // Reversible blocks are *usually* in the TAPOS buffer.  Since this
+      // is the fastest check, we do it first.
+      block_summary_id_type bsid = block_num & 0xFFFF;
```

Comment:
The assumption that TaPoS is exactly 16 bits is already baked into various places in the code, e.g. [here](https://github.com/steemit/steem/blob/e6ef4ebdb5cd160b71f70526deb8273401d6c271/libraries/protocol/transaction.cpp#L72).

Making the mask a configurable constant means you'd have to review / update the code in all places that would need to be changed to make it a flexible value.  Which is so far off-topic, I'd say it should be a new ticket entirely.